src/qtxdg/xdgmimetype.h

```diff
@@ -72,12 +72,12 @@ class QTXDG_API XdgMimeType : public QMimeType {
      */
     bool operator==(const XdgMimeType &other) const
     {
-        return QMimeType::operator==(other);
+        return name() == other.name();
```

Comment:
Why dropping the `QMimeType` logic? IMO we can be more safe for future Qt changes, if we continue to stick with Qt's implementation:
```cpp
return static_cast<const QMimeType &>(*this) == other;
```

Reply:
> Why dropping the QMimeType logic?

Basic things have changed in Qt 6.8 (also, see https://github.com/lxqt/libqtxdg/issues/306#issuecomment-2348995325). Moreover, two mimetypes are equal if their names are.